File: 3336-water-bottles-ii/3336-water-bottles-ii.cpp
class Solution {
public:
    int maxBottlesDrunk(int numBottles, int numExchange) {
        int res = numBottles;
        int empty = numBottles;
        while(empty >= numExchange)
        {
            res++;
            empty -= numExchange - 1;
            numExchange++;
        }
        return res;
    }
};

File: 3336-water-bottles-ii/3336-water-bottles-ii.py
class Solution:
    def maxBottlesDrunk(self, numBottles: int, numExchange: int) -> int:
        res = numBottles
        empty = res
        while empty >= numExchange:
            res += 1
            empty -= numExchange - 1
            numExchange += 1
        return res